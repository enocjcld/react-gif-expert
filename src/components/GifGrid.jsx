import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { images, isLoading } =  useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>
            {
                // if isLoading es true haz cargando
                isLoading && ( <h2>Cargando...</h2> )
            }

            <div className="card-grid">
                {/* images.map */}
                { 
                    images.map( ( image ) => (
                        <GifItem
                            key={ image.id }
                            { ...image } // Esto es exparcion de properties "recibe del componete gifItem"
                        />
                    )) 
                }
            </div>

        </>
    )
}
