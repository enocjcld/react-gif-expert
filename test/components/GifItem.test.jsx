import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Prueba del GifItem', () => {

    const title = 'Saitama';
    const url   = 'https://one-punch.com/saitama.jpg';

    test('debe de hacer match con los snapshops', () => {
        
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('debe de mistrar la imagen con el URL y el ALT indicado', () => {

        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('debe mostrar el titulo en el componente', () => {

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title) ).toBeTruthy();
    });
});