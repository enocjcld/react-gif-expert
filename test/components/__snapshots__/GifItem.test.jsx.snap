// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del GifItem debe de hacr match con los snapshops 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
